loader/integration.go

```diff
@@ -55,3 +60,93 @@ func GetLatestPipelineRunForSnapshotAndScenario(adapterClient client.Client, ctx
 
 	return nil, err
 }
+
+// prepareSnapshot prepares the Snapshot for a given application and the updated component (if any).
+// In case the Snapshot can't be created, an error will be returned.
+func PrepareSnapshot(adapterClient client.Client, ctx context.Context, loader ObjectLoader, application *applicationapiv1alpha1.Application, component *applicationapiv1alpha1.Component, newContainerImage string, newComponentSource *applicationapiv1alpha1.ComponentSource) (*applicationapiv1alpha1.Snapshot, error) {
```

Comment:
I'm not sure if `PrepareSnapshot` should go under the `loader` package, maybe [gitops/snapshot.go](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go) would make more sense?

Reply:
Because `prepareSnapshot` calls function `GetAllApplicationComponents` from `loader` package, if I add `prepareSnapshot` to `gitops`  need to import `loader`, I am afraid we will have import conflict. But let me try it. Thanks.

Reply:
If that doesn't work, we can refactor the `PrepareSnapshot()` to take a list of ApplicationComponents instead of [fetching them within the function](https://github.com/redhat-appstudio/integration-service/blob/1a4175c9c8bc633f3489868198982b99fb0342b7/controllers/pipeline/pipeline_adapter.go#L414) - that way the loader function would not be used within it, but from the adapters, wdyt?

Reply:
This is a good idea.

Reply:
Maybe I can also change [integrationPipelineRuns](https://github.com/redhat-appstudio/integration-service/blob/1a4175c9c8bc633f3489868198982b99fb0342b7/loader/integration.go#L33) as a parameter and move [GetLatestPipelineRunForSnapshotAndScenario()](https://github.com/redhat-appstudio/integration-service/blob/1a4175c9c8bc633f3489868198982b99fb0342b7/loader/integration.go#L32) out of `loader` and back to the older package. Last time, due to the same reason, I move it to `loader` package?  @dirgim

Reply:
That sounds reasonable to me.

Reply:
It is on the edge of expanding the scope for this PR, but since it's a function that's called from the [current pipeline adapter](https://github.com/redhat-appstudio/integration-service/blob/1a4175c9c8bc633f3489868198982b99fb0342b7/controllers/pipeline/pipeline_adapter.go#L390) which is being refactored, I think it can be included.

Reply:
Thanks. So let me move [GetLatestPipelineRunForSnapshotAndScenario()](https://github.com/redhat-appstudio/integration-service/blob/1a4175c9c8bc633f3489868198982b99fb0342b7/loader/integration.go#L32) in a new PR, otherwise this PR will be bigger and bigger and not easy to review.

Reply:
`PrepareSnapshot()` is moved to the snapshot.go under `gitops` pkg, thanks!